app/services/vot/acr_component_values.rb

```diff
@@ -1,105 +1,302 @@
+      IALMAX,
+      IAL1,
+      LOA1,
+    ].freeze
 
+    # IAL names ordered by priority, descending
+    IALS_BY_PRIORITY_LIST = IAL_COMPONENTS_BY_PRIORITY_LIST.map(&:name).freeze
+
+    # AAL components ordered by priority, descending
+    AAL_COMPONENTS_BY_NAME_HASH = [
+      DEFAULT_AAL,
+      AAL1,
+      AAL2,
+      AAL2_PHISHING_RESISTANT,
+      AAL2_HSPD12,
+      AAL3,
+      AAL3_HSPD12,
+    ].index_by(&:name).freeze
+
+    # AAL components ordered by priority, descending
+    AAL_COMPONENTS_BY_PRIORITY_LIST = [
+      AAL2_HSPD12,
+      AAL3_HSPD12,
+      AAL2_PHISHING_RESISTANT,
+      AAL3,
+      AAL2,
+      AAL1,
+      DEFAULT_AAL,
+    ].freeze
+
+    # AAL names ordered by priority, descending
+    AALS_BY_PRIORITY_LIST = AAL_COMPONENTS_BY_PRIORITY_LIST.map(&:name).freeze
+
+    # All ACR components ordered by type (IAL, AAL) and then descending priority
+    ACR_COMPONENTS_BY_PRIORITY_LIST = [
+      *IAL_COMPONENTS_BY_PRIORITY_LIST,
+      *AAL_COMPONENTS_BY_PRIORITY_LIST,
+    ].freeze
+
+    ACRS_BY_PRIORITY_LIST = ACR_COMPONENTS_BY_PRIORITY_LIST.map(&:name).freeze
+
+    # @return [Hash{String=>AcrComponentValue}]
     def self.by_name
       NAME_HASH
     end
-  end
+
+    def self.satisfies?(name, *requirements)
+      component = NAME_HASH[name]
+
+      component.present? &&
+        component.requirements.intersection(requirements).size == requirements.length
+    end
+
+    def self.highest_priority_ial(values)
+      ial_component_values(values).min&.name
+    end
+
+    def self.highest_priority_aal(values)
+      aal_component_values(values).min&.name
+    end
+
+    # Get the highest priority ACR value
+    # @return [String, nil]
+    def self.find_highest_priority(values)
+      AcrComponentValues.order_by_priority(values).first
+    end
+
+    # Sort ACR values by priority, highest to lowest
+    # @param values [Array<String,AcrComponentValue>, String]
+    def self.order_by_priority(values)
+      order_by_priority_with(values)
+    end
+
+    # Order a list of ACR values by priority, highest to lowest
+    # Returns a new {Array} of {String} values where the order has been by set by the +series+,
+    # based on the index of the objects from the original  in the series.
+    #
+    # If the +series+ includes values that have no corresponding element in the Enumerable,
+    # these are ignored.
+    # If the Enumerable has additional elements that aren't named in the +series+,
+    # these are not included in the result.
+    # @param values [Array<String, AcrComponentValue>, String]
+    # @param series [Array<String>,nil] Defaults to #ACRS_BY_PRIORITY_LIST
+    def self.order_by_priority_with(values, series: nil)
+      if series.present?
+        to_names(values).
+          filter { |acr| AcrComponentValues.acr?(acr) && series.presence.include?(acr) }.
+          sort_by { |acr| series.presence.index(acr) }
+      else
+        to_components(values).sort.map(&:name)
+      end
+    end
```

Comment:
This whole method _could_ go away. It is a leftover of `ComponentValue` not implementing `Comparable`.